src/components/apps/launch/AppLaunchForm.js

```diff
@@ -0,0 +1,600 @@
+                paramType === constants.PARAM_TYPE.MULTIFILE_SELECTOR
+                    ? defaultValue || []
+                    : paramArgs?.length > 0
+                    ? paramArgs.find(
+                          (arg) => arg.isDefault || defaultValue?.id === arg.id
+                      )
+                    : defaultValue || "";
+
+            return {
+                ...param,
+                value,
+            };
+        }),
+    }));
+
+    const reqInitValues = requirements?.map(
+        ({
+            step_number,
+            default_cpu_cores = 0,
+            default_memory = 0,
+            default_disk_space = 0,
+        }) => ({
+            step_number,
+            min_cpu_cores: default_cpu_cores,
+            min_memory_limit: default_memory,
+            min_disk_space: default_disk_space,
+        })
+    );
+
+    return {
+        debug: false,
+        notify,
+        output_dir: defaultOutputDir,
+        name: formatAnalysisName(intl, name),
+        description: "",
+        app_id: id,
+        system_id,
+        groups: groupInitValues,
+        limits: requirements,
+        requirements: reqInitValues || [],
+    };
+};
+
+/**
+ * Formats the analysis submission from props and the form values.
+ *
+ * @param {string} defaultOutputDir
+ * @param {Object} formValues
+ * @param {boolean} formValues.notify
+ * @param {boolean} formValues.debug
+ * @param {string} formValues.name
+ * @param {string} formValues.description
+ * @param {string} formValues.output_dir
+ * @param {string} formValues.system_id
+ * @param {string} formValues.app_id
+ * @param {Object[]} formValues.requirements
+ * @param {Object[]} formValues.groups
+ *
+ * @returns The formatted submission for launching or saving as a quick launch.
+ */
+const formatSubmission = (
+    defaultOutputDir,
+    {
+        notify,
+        debug,
+        name,
+        description,
+        output_dir,
+        system_id,
+        app_id,
+        requirements,
+        groups,
+    }
+) => ({
+    notify,
+    debug,
+    create_output_subdir: output_dir === defaultOutputDir,
+    name: name.trim(),
+    description,
+    output_dir,
+    system_id,
+    app_id,
+    requirements,
+    config: groups?.reduce(paramConfigsReducer, {}),
+});
+
+/**
+ * Appends the given group's parameter values to the given submission config.
+ *
+ * @param {Object} configs - Mapping of parameter IDs to parameter values.
+ * @param {Object} group
+ * @param {Object[]} group.parameters
+ *
+ * @returns The formatted submission config
+ * including the given group's parameter IDs to values.
+ */
+const paramConfigsReducer = (configs, group) => {
+    group.parameters.forEach((param) => {
+        const { id, type } = param;
+
+        if (type !== constants.PARAM_TYPE.INFO) {
+            let { value } = param;
+
+            switch (type) {
+                case constants.PARAM_TYPE.FLAG:
+                    value = value && value !== "false";
```

Comment:
I'm using the [FormCheckboxStringValue](https://github.com/cyverse-de/ui-lib/blob/2f9df43cf68b79baf1dd3f85a21710964b5b9273/src/util/FormField.js#L124-L138), which stores a string value of "true" or "false" in the form, rather than a literal boolean. I'm using the string values so I don't have to do any conversions to display the selected value in the Review step, but that means converting the string value to a literal boolean for the submission configs here.

If it seems clearer, I could swap these fields to use a regular [FormCheckbox](https://github.com/cyverse-de/ui-lib/blob/2f9df43cf68b79baf1dd3f85a21710964b5b9273/src/util/FormField.js#L89-L106), and convert the boolean values to display strings in the Review step instead. Maybe that's even better, so we can choose what string to display in the review step if we want something other than "true" or "false" 🤔

Reply:
Ah, I see.  I didn't notice the usage of `FormCheckboxStringValue`, so I was mostly confused about that.  I'll leave it up to you if you want to switch.  Now that I'm aware of it, it makes sense, so leaving it is fine too 👍